src/software/geom/util.cpp

```diff
@@ -899,85 +903,79 @@ std::optional<Point> getRayIntersection(Ray ray1, Ray ray2)
 
 Vector reflect(const Vector &v, const Vector &n)
 {
-    if (n.len() < EPS)
+    if (n.length() < EPS)
     {
         return v;
     }
-    Vector normal = n.norm();
+    Vector normal = n.normalize();
     return v - 2 * v.dot(normal) * normal;
 }
 
-Vector reflect(const Vector &a, const Vector &b, const Vector &p)
+Point reflect(const Point &a, const Point &b, const Point &p)
 {
     // Make a as origin.
     // Rotate by 90 degrees, does not matter which direction?
     Vector n = (b - a).rotate(Angle::quarter());
     return a + reflect(p - a, n);
 }
 
-Vector calcBlockCone(const Vector &a, const Vector &b, const double &radius)
+Point calcBlockCone(const Vector &a, const Vector &b, const double &radius)
 {
-    if (a.len() < EPS || b.len() < EPS)
+    if (a.length() < EPS || b.length() < EPS)
```

Comment:
How was the `EPS` determined for geometry utilities? We now have an `EPSILON` defined in `geom_constants.h`, but it has a value much smaller than `EPS`. Either way, I think it would be nice to have all constants in `geom_constants.h`. @MathewMacDougall @garethellis0

Reply:
Here's a good article on what epsilon is and how to determine it: https://en.wikibooks.org/wiki/Floating_Point/Epsilon

Not sure why I never thought to look for it before, but C++ has a `epsilon` in the standard library: https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon, perhaps we should just use this instead?

Reply:
Oh neat didn't realize `numeric_limits` had an `epsilon`. Don't know why `geom/util` has its own `EPS` value.

Reply:
So the `epsilon` in standard library is `2.22045e-16`, which is quite a bit smaller than `EPSILON` as defined in `geom_constants.h`. According to its comments: `1e-15 was chosen as a value because doubles have about 16 consistent significant figures. Comparing numbers with 15 significant figures gives us a small buffer while remaining as accurate as possible.` I'm all for using a constant defined by the standard library, but concerned that the EPSILON value provided might be too small. Thoughts?

Reply:
Hmm, good point. Let's chat about this at meeting tmw.

Reply:
Chatted about it and we now have this issue #1023 feel free to edit if I missed anything